Fix locally created database

amazari opened the pull request:
###### Motivation for this change
In its current form the database.createLocally flag is broken. Those patches try to fix it.

###### Things done
 - mysql user auth: ensure that the mysql and the unix user running zonminder share the same name in order to allow unix socket mysql connection as enforced by the 'ensureUsers' directive
 - move the initial db creation and population to the right scope
Ping @peterhoeg 

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

amazari committed:
zoneminder: Fix mysql username to match unix username

When database.createLocally is used, a mysql user is created with the ensureUsers directive.
It ensures that the unix user with the name provided exists and can connect to MySQL through socket.
Thus, the MySQL username used by php/perl scripts must match the unix user owning the server PID.

This patch sets the default mysql to 'zoneminder' instead of 'zmuser'.

amazari committed:
zoneminder: fix initial database creation

Move initialDatabases directive from the 'ensureUsers' scope to the correct outer 'mysql' one.

danbst commented on nixos/modules/services/misc/zoneminder.nix:
```diff
@@ -49,7 +49,7 @@ let
     ZM_DB_TYPE=mysql
     ZM_DB_HOST=${cfg.database.host}
-    ZM_DB_NAME=${cfg.database.name}
+    ZM_DB_NAME=${if cfg.database.createLocally user else cfg.database.name}
```
There is syntax problem here - `then` is missing. I guess this particular change wasn't tested somehow.

amazari replied:
Yup, sorry about that. Will fix that asap, thanks for looking into it.

amazari committed:
zoneminder: database.createLocally fix

Fix syntax error and use the correct variable (cfg.database.username) when not creating the db locally.

amazari commented:
Thanks @danbst for the merge.
Sadly I ***ed up and made a stupid mistake. It is fixed in #57125 . Would you mind reviewing it so master doesn't stay broken for too long ?
Sorry for that!

JohnAZoidberg commented on nixos/modules/services/misc/zoneminder.nix:
```diff
@@ -49,7 +49,7 @@ let
     ZM_DB_TYPE=mysql
     ZM_DB_HOST=${cfg.database.host}
-    ZM_DB_NAME=${cfg.database.name}
+    ZM_DB_NAME=${if cfg.database.createLocally then user else cfg.database.username}
```
Why? Why would you want a different user when having zoneminder create the database?
